src/transition.S

```diff
@@ -40,9 +40,40 @@ comp_exec_in:
     stp lr, x29, [sp, #-16]!
     mov x19, sp
 
-    mov sp, x0
-    msr DDC, c1
-    blr x2
+    // Move arguments to temporary registers (we'll need them later)
+    mov x9, x0
+    mov c10, c1
+    mov x11, x2
+
+    // Load params (only handle 3 params):
+loading_params:
+    cbz x4, loaded_params
+    ldr x0, [x3], #8
+    ldr x0, [x0]
```

Comment:
Why is there double indirection here? It'd be simpler (and more efficient) to pass by value, at least within the `args` array. For example, `get_entry_point_args` could avoid the `malloc` in the inner loop.

I suspect that much of this will change if you add float or capability support anyway, so it's not important to address now.